test/e2e-preferred-host/ks_preferred_host_kas_test.go

```diff
@@ -1,7 +1,188 @@
 package e2e_preferred_host
 
-import "testing"
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/stretchr/testify/require"
+	"math/rand"
+	"net"
+	"net/url"
+	"strconv"
+	"testing"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apimachinery/pkg/watch"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	watchtools "k8s.io/client-go/tools/watch"
+
+	configv1client "github.com/openshift/client-go/config/clientset/versioned"
+	operatorv1client "github.com/openshift/client-go/operator/clientset/versioned/typed/operator/v1"
+	"github.com/openshift/cluster-kube-scheduler-operator/pkg/operator/operatorclient"
+	test "github.com/openshift/cluster-kube-scheduler-operator/test/library"
+	testlibrary "github.com/openshift/library-go/test/library"
+)
+
+var (
+	// the following parameters specify for how long apis must
+	// stay on the same revision to be considered stable
+	waitForKSORevisionSuccessThreshold = 3
+	waitForKSORevisionSuccessInterval  = 1 * time.Minute
+
+	// the following parameters specify max timeout after which
+	// KSOs are considered to not converged
+	waitForKSORevisionPollInterval = 30 * time.Second
+	waitForKSORevisionTimeout      = 7 * time.Minute
+
+	// podProvisionTimeout is how long to wait for a service account to be provisioned.
+	// service accounts are provisioned after namespace creation
+	// a service account is required to support pod creation in a namespace as part of admission control
+	podProvisionTimeout = 2 * time.Minute
+
+	nsCreationPollInterval = 2 * time.Second
+)
 
 func TestKSTalksOverPreferredToKas(t *testing.T) {
-	t.Log("implement me")
+	// test data
+	kubeConfig, err := test.NewClientConfigForTest()
+	require.NoError(t, err)
+	kubeClient, err := clientset.NewForConfig(kubeConfig)
+	require.NoError(t, err)
+	operatorClientSet, err := operatorv1client.NewForConfig(kubeConfig)
+	require.NoError(t, err)
+	ksoOperator := operatorClientSet.KubeSchedulers()
+	ksoOperatorConfig, err := ksoOperator.Get(context.TODO(), "cluster", metav1.GetOptions{})
+	require.NoError(t, err)
+
+	openShiftConfigClient, err := configv1client.NewForConfig(kubeConfig)
+	require.NoError(t, err)
+
+	t.Log("getting the current Kube API host")
+	scheme, host := readCurrentKubeAPIHostAndScheme(t, openShiftConfigClient)
+
+	t.Logf("setting the \"unsupported-kube-api-over-localhost\" flag and pointing the current master to %q (non available host)", fmt.Sprintf("%s://%s:1234", scheme, host))
+	data := map[string]map[string][]string{
+		"extendedArguments": {
```

Comment:
you need to replace it with something like that:
```
unsupportedConfigOverrides:
  arguments:
    unsupported-kube-api-over-localhost:
      - "true"
    master:
      - "abc"
```


we added support in https://github.com/openshift/cluster-kube-scheduler-operator/pull/336/commits/e837452b29cd52b93748017b46bda4274a775260#r600414724